File: Unit4.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-24
 * @fileoverview This program prints out the times table.
 */

// variables
let numberAsString1: string = "";
let numberAsNumber1: number = 0;

// input number
numberAsString1 = prompt("Enter a number (1-12): ") || "0";
numberAsNumber1 = parseInt(numberAsString1);

// print times table
console.log(`\nTimes Table for ${numberAsNumber1}:`);
for (let counter: number = 1; counter <= 12; counter++) {
  console.log(`${numberAsNumber1} x ${counter} = ${numberAsNumber1 * counter}`);
}

console.log("\nDone.");

File: Unit4.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-11-24
// Fileoverview: This program prints out the times table.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	// variable declaration
	var numberAsString string
	var numberAsNumber int

	var reader = bufio.NewReader(os.Stdin)

	// input number
	fmt.Print("Enter a number (1-12): ")
	numberAsString, _ = reader.ReadString('\n')
	numberAsString = strings.TrimSpace(numberAsString)
	numberAsNumber, _ = strconv.Atoi(numberAsString)

	// print times table
	fmt.Println("\nTimes Table for", numberAsNumber)
	for counter := 1; counter <= 12; counter++ {
		fmt.Printf("%d x %d = %d\n", numberAsNumber, counter, numberAsNumber*counter)
	}

	fmt.Println("\nDone.")
}
